core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -137,14 +137,36 @@ public void run() {
                     jenkins.trimLabels();
                 }
             });
-            // no need for a full save() so we just do the minimum
-            if (node instanceof EphemeralNode) {
-                Util.deleteRecursive(new File(getNodesDir(), node.getNodeName()));
-            } else {
-                XmlFile xmlFile = new XmlFile(Jenkins.XSTREAM,
-                        new File(new File(getNodesDir(), node.getNodeName()), "config.xml"));
-                xmlFile.write(node);
-            }
+            persistNode(node);
+        }
+    }
+
+    /**
+     * Actually persists a node on disk.
+     *
+     * @param node the node to be persisted.
+     * @throws IOException if the node could not be persisted.
+     */
+    private void persistNode(final @Nonnull Node node)  throws IOException {
+        // no need for a full save() so we just do the minimum
+        if (node instanceof EphemeralNode) {
+            Util.deleteRecursive(new File(getNodesDir(), node.getNodeName()));
+        } else {
+            XmlFile xmlFile = new XmlFile(Jenkins.XSTREAM,
+                    new File(new File(getNodesDir(), node.getNodeName()), "config.xml"));
+            xmlFile.write(node);
+        }
+    }
+
+    /**
+     * Updates an existing node on disk.
+     *
+     * @param node the node to be updated.
+     * @throws IOException if the node could not be persisted.
+     */
+    public void updateNode(final @Nonnull Node node) throws IOException {
+        if (node == nodes.get(node.getNodeName())) {
+            persistNode(node);
```

Comment:
sorry didn't notice this earlier - but this should at least log if a call was made and the `if` statement results to false.
I'm wondering if this should be more than a log as well and actually throw an exception.